api/internal/handler/story_handler.go

```diff
@@ -165,15 +166,31 @@ func NewAdminCreateStoryChapterHandler(
 		}
 
 		storyId := storyInfoResult.ID
-		storyChapter := entity.ToStoryChapterModel(&requestBody, storyId, chapterNum)
-		err = storyChapterRepo.CreateChapter(storyChapter)
+		chapter, err := storyChapterRepo.GetChapter(storyId, chapterNum)
+		if err == gorm.ErrRecordNotFound {
+			chapter = entity.ToStoryChapterModel(&requestBody, storyId, chapterNum)
+			err = storyChapterRepo.CreateChapter(chapter)
+			if err != nil {
+				logger.Errorf("Failed to create story chapter: %v", err)
+				c.JSON(http.StatusInternalServerError, ErrorMessage("Internal server error"))
+				return
+			}
+		}
+		if err != nil {
+			logger.Errorf("Failed to get story chapter: %v", err)
+			c.JSON(http.StatusInternalServerError, ErrorMessage("Internal server error"))
+			return
+		}
+
+		chapter.FromCreateStoryChapterRequestBody(&requestBody)
```

Comment:
always check if chapter is nil to be safe

Reply:
updated